akka-actor-tests/src/test/scala/akka/actor/DeathWatchSpec.scala

```diff
@@ -70,6 +71,42 @@ object DeathWatchSpec {
   final case class FF(fail: Failed)
 
   final case class Latches(t1: TestLatch, t2: TestLatch) extends NoSerializationVerificationNeeded
+
+  object WatchWithVerifier {
+    case class WatchThis(ref: ActorRef)
+    case object Watching
+    case class CustomWatchMsg(ref: ActorRef)
+    case class StartStashing(numberOfMessagesToStash: Int)
+    case object StashingStarted
+
+    def props(probe: ActorRef) = Props(new WatchWithVerifier(probe))
+  }
+  class WatchWithVerifier(probe: ActorRef) extends Actor with Stash {
+    import WatchWithVerifier._
+    private var stashing = false
+    private var stashNMessages = 0
+
+    override def receive: Receive = {
+      case StartStashing(messagesToStash) =>
+        stashing = true
+        stashNMessages = messagesToStash
+        sender ! StashingStarted
+      case WatchThis(ref) =>
+        context.watchWith(ref, CustomWatchMsg(ref))
+        sender ! Watching
+      case _ if stashing =>
+        stash()
+        stashNMessages -= 1
+        if (stashNMessages == 0) {
+          stashing = false
+          unstashAll()
+        }
+      case msg @ CustomWatchMsg(_) =>
+        probe ! msg
+      case msg @ Terminated(_) =>
+        probe ! msg
```

Comment:
isn't that the classical problem that `Terminated` can't be forwarded to a probe? Wrap in something else.

Reply:
Do'h! We don't actually need it at all, I'll remove it.